lib/iris/tests/unit/combine/test_CombineOptions.py

```diff
@@ -0,0 +1,215 @@
+        In the case of .context(), the 'after' is within the block, and the 'before'
+        state should always be restored again afterwards.
+        """
+        if initial_options is not None:
+            options = initial_options
+        else:
+            options = CombineOptions()
+
+        op_kwargs = op_kwargs or {}
+
+        if before_checks is not None:
+            assert options_checks(options, before_checks)
+
+        if op_is_set:
+            # do "set" check
+            options.set(op_arg, **op_kwargs)
+            assert options_checks(options, after_checks)
+        else:
+            # do "context" checks
+            with options.context(op_arg, **op_kwargs):
+                assert options_checks(options, after_checks)
+            assert options_checks(options, before_checks)
+
+    @pytest.fixture(params=["set", "context"])
+    def op_is_set(self, request):
+        """Parametrise a test over both .set() and and .context() calls."""
+        return request.param == "set"
+
+    def test_empty_set(self):
+        # More or less, just check that an empty set() call is OK.
+        options = CombineOptions()
+        orig_settings = options.settings()
+        options.set()
+        assert options.settings() == orig_settings
+
+    def test_empty_context(self):
+        # More or less, just check that an empty context() call is OK.
+        options = CombineOptions()
+        orig_settings = options.settings()
+        with options.context():
+            assert options.settings() == orig_settings
+
+    def test_arg_dict(self, op_is_set):
+        expect_before = {"merge_concat_sequence": "m", "repeat_until_unchanged": False}
+        set_arg = {"merge_concat_sequence": "c", "repeat_until_unchanged": True}
+        expect_after = {"merge_concat_sequence": "c", "repeat_until_unchanged": True}
+        self.do_check(
+            op_arg=set_arg,
+            before_checks=expect_before,
+            after_checks=expect_after,
+            op_is_set=op_is_set,
+        )
+
+    def test_arg_string(self, op_is_set):
+        expect_before = {"merge_concat_sequence": "m", "repeat_until_unchanged": False}
+        set_arg = "comprehensive"
+        expect_after = {"merge_concat_sequence": "mc", "repeat_until_unchanged": True}
+        self.do_check(
+            op_arg=set_arg,
+            before_checks=expect_before,
+            after_checks=expect_after,
+            op_is_set=op_is_set,
+        )
+
+    def test_kwargs(self, op_is_set):
+        expect_before = {"merge_concat_sequence": "m", "repeat_until_unchanged": False}
+        set_arg = {"merge_concat_sequence": "c", "repeat_until_unchanged": True}
+        expect_after = {"merge_concat_sequence": "c", "repeat_until_unchanged": True}
+        self.do_check(
+            op_arg=set_arg,
+            before_checks=expect_before,
+            after_checks=expect_after,
+            op_is_set=op_is_set,
+        )
+
+    def test_arg_kwargs(self, op_is_set):
```

Comment:
I think it's worth adding a similar test for when the arg is a string.

Reply:
Yes I see, nice spot !
[0c11048](https://github.com/SciTools/iris/pull/6334/commits/0c1104890cc80bd3efc7b446f1492f1bda6e3103)

NOTE: I found I'm not actually using the "initial_options" arg of the `do_check` method any more, as all the current tests are in fact compatible with starting from a "default" state.  But I have left it in for now.
If you think this should be removed, I will do so.